### ISSUE TITLE
NPE: double com.mapbox.mapboxsdk.camera.CameraPosition.zoom

### ISSUE THREAD

**Issue by hrach** [mapbox/mapbox-navigation-android#1969]
### Steps to reproduce

Not sure. 

### Expected behavior

No crash.

### Actual behavior

```
Fatal Exception: java.lang.NullPointerException
Attempt to read from field 'double com.mapbox.mapboxsdk.camera.CameraPosition.zoom' on a null object reference
com.mapbox.services.android.navigation.ui.v5.camera.DynamicCamera.createZoom (DynamicCamera.java:110)
com.mapbox.services.android.navigation.ui.v5.camera.DynamicCamera.zoom (DynamicCamera.java:60)
com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera.adjustCameraForReset (NavigationCamera.java:526)
com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera.onCameraTransitionFinished (NavigationCamera.java:432)
com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera.updateTransitionListenersFinished (NavigationCamera.java:352)
com.mapbox.services.android.navigation.ui.v5.camera.NavigationCameraTransitionListener.onLocationCameraTransitionFinished (NavigationCameraTransitionListener.java:15)
com.mapbox.mapboxsdk.location.LocationComponent$CameraTransitionListener.onLocationCameraTransitionFinished (LocationComponent.java:572)
com.mapbox.mapboxsdk.location.LocationCameraController$1.onFinish (LocationCameraController.java:135)
com.mapbox.mapboxsdk.maps.Transform$2.run (Transform.java:89)
```

### Configuration

**Android versions:** 6, 8, 9
**Device models:** Various
**Mapbox SDK versions:** 
```
com.mapbox.mapboxsdk:mapbox-android-sdk:7.4.0
com.mapbox.mapboxsdk:mapbox-android-navigation:0.38.0
com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.38.0
```

**Comment by tobrun** [comment]
@hrach thank you for reaching out, moving this issue to `mapbox-navigation-android` as this nullpointer originates from this repository.

**Comment by Guardiola31337** [comment]
@hrach Thanks for reaching out and report your issue.

Fix will be included in the next release `v0.41.0` 👀 https://github.com/mapbox/mapbox-navigation-android/pull/1979

That being said and as described in the PR https://github.com/mapbox/mapbox-navigation-android/pull/1979#issue-289899630 this NPE isn't reproducible if lifecycle methods from `NavigationCamera` https://github.com/mapbox/mapbox-navigation-android/blob/410672dfc69cfa95296b08c4f8bdf2c14bcfde88/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/NavigationCamera.java#L275-L295 or `NavigationMapboxMap` https://github.com/mapbox/mapbox-navigation-android/blob/410672dfc69cfa95296b08c4f8bdf2c14bcfde88/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java#L461-L483 are called as part of the `Activity` / `Fragment` lifecycle. You can 👀 [`ComponentNavigationActivity`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/ComponentNavigationActivity.java) for further details on how to use these UI components.

Are you using any of those UI components by any chance? If so, in order to prevent unexpected behaviors you should make sure that their lifecycle methods are called - internally, we're cleaning up a bunch of resources (listeners, etc.) making sure that non-UI methods are executed and preventing these kind of NPEs.

cc @andrlee

**Comment by hrach** [comment]
Hi, I'm doing something like this (in fragment)

```kotlin
viewModel.navigation.observe(this) { (navigation, directions) ->
     val navigationCamera = NavigationCamera(mapboxMap, navigation, mapboxMap.locationComponent)
     this.lifecycle.addObserver(navigationCamera)
     navigationCamera.start(directions)
}
```
